File: LeetCode/Java/Easy/MinimumDepthOfBinaryTree.java
package Easy;

public class MinimumDepthOfBinaryTree {

	public static void main(String[] args) {

	}
	
	public static int minDepth(TreeNode root){
		if (root == null){
			return 0;
		}
		return minDepthHelper(root, 1);
	}

	public static int minDepthHelper(TreeNode node, int depth){
		
		if (node.right == null && node.left == null){
			return depth;
		}
		if (node.right == null && node.left != null){
			return minDepthHelper(node.left, depth+1);
		}
		if (node.right != null && node.left == null){
			return minDepthHelper(node.right, depth+1);
		}
		else {
			return Math.min(minDepthHelper(node.right, depth+1), minDepthHelper(node.left, depth+1));
		}
		
		
	}
}

File: LeetCode/Python/Easy/MinimumDepthOfBinaryTree.py
class Solution(object):
    def minDepth(self, root):
        if root is None:
            return 0

        return self.minDepthHelper(root)

    def minDepthHelper(self, root):
        if root.left is None and root.right is None:
            return 1

        left = self.minDepthHelper(root.left) if root.left is not None else float('inf')
        right = self.minDepthHelper(root.right) if root.right is not None else float('inf')

        return min(left, right) + 1
